src/accel/LocalTransporter.cc

```diff
@@ -191,6 +193,23 @@ void LocalTransporter::Push(G4Track const& g4track)
 {
     CELER_EXPECT(*this);
 
+    if (Real3 pos = convert_from_geant(g4track.GetPosition(), 1);
+        !is_inside(bbox_, pos))
```

Comment:
I wonder if (for our `UserTrackingAction` integration at least) we could prevent this by checking that the `G4TrackStatus` is valid before offloading in the pre tracking action? Looking at how Geant4 handles tracks outside the geometry, in the stepping manager `SetInitialStep` [if a track is outside](https://github.com/Geant4/geant4/blob/32390e802bd84a8b85ac02a90a3d31cf205dd5b7/source/tracking/src/G4SteppingManager.cc#L348-L362) an error is thrown if it's a primary and the track status is set to `fStopAndKill` if it's a secondary. The `PreUserTrackingAction` is called just after this.

Not sure about the tracking manager integration though, I don't think the volume is checked or status changed before handing over the track...

Reply:
> Not sure about the tracking manager integration though, I don't think the volume is checked or status changed before handing over the track...

Indeed, it looks like the status stays the same with the tracking manager, a printout for outside shows that they are marked as alive:

```
LocalTransporter.cc:202: debug: Discarding track outside world: 0.00119591 MeV from e- at {1990.41,-1141.8,-6566.06} along {-0.181364,-0.833315,-0.5222}
LocalTransporter.cc:216: debug: Track status: fAlive
```